tools/points-service/main.go

```diff
@@ -0,0 +1,529 @@
+	rwLock.RLock()
+	defer rwLock.RUnlock()
+
+	var existingAdder string
+	err := db.QueryRow(`
+        SELECT adder FROM events
+        WHERE pubkey = ? AND opted_out_block IS NULL
+    `, pubkey).Scan(&existingAdder)
+
+	// if pubkey is active under a different adder, ignore
+	if err == nil && existingAdder != "" && existingAdder != adder {
+		logger.Warn("pubkey already opted in by a different adder",
+			"pubkey", pubkey, "existing_adder", existingAdder, "new_adder", adder)
+		return
+	}
+
+	_, err = db.Exec(`
+        INSERT INTO events (
+            pubkey, adder, registry_type, event_type, 
+            opted_in_block, opted_out_block, points_accumulated
+        ) VALUES (?, ?, ?, ?, ?, NULL, 0)
+    `, pubkey, adder, registryType, eventType, inBlock)
+	if err != nil {
+		logger.Warn("insertOptIn likely already inserted", "error", err)
+	} else {
+		logger.Info("inserted opt-in interval",
+			"pubkey", pubkey, "block", inBlock, "event_type", eventType, "adder", adder)
+	}
+}
+
+// Mark an existing interval as opted-out
+func insertOptOut(db *sql.DB, logger *slog.Logger, pubkey, adder, eventType string, outBlock uint64) {
+	rwLock.RLock()
+	defer rwLock.RUnlock()
+
+	_, err := db.Exec(`
+        UPDATE events
+        SET opted_out_block = ?, event_type = ?
+        WHERE pubkey = ? AND adder = ? AND opted_out_block IS NULL
+    `, outBlock, eventType, pubkey, adder)
+	if err != nil {
+		logger.Error("failed to opt-out", "error", err, "pubkey", pubkey)
+	} else {
+		logger.Info("opt-out interval updated",
+			"pubkey", pubkey, "block", outBlock, "event_type", eventType, "adder", adder)
+	}
+}
+
+// --------------------------------------
+//   4) MAIN
+// --------------------------------------
+
+func main() {
+	app := &cli.App{
+		Name:  "mev-commit-points",
+		Usage: "MEV Commit Points Service",
+		Flags: []cli.Flag{optionRPCURL},
+		Action: func(c *cli.Context) error {
+			logger := slog.New(slog.NewJSONHandler(os.Stdout, nil))
+
+			// 1. Connect to DB
+			db, err := initDB(logger)
+			if err != nil {
+				return fmt.Errorf("failed to connect to database: %w", err)
+			}
+			defer db.Close()
+
+			// 2. Test DB
+			var rowCount int
+			err = db.QueryRow("SELECT COUNT(*) FROM events").Scan(&rowCount)
+			if err != nil {
+				return fmt.Errorf("failed to query events: %w", err)
+			}
+			logger.Info("database reachable", "events_count", rowCount)
+
+			// 3. Ethereum client
+			ethClient, err := ethclient.Dial(c.String(optionRPCURL.Name))
+			if err != nil {
+				return fmt.Errorf("failed to connect to Ethereum node: %w", err)
+			}
+
+			// 4. Load ABIs
+			contractABIs, err := getContractABIs()
+			if err != nil {
+				return fmt.Errorf("failed to get contract ABIs: %w", err)
+			}
+			listener := events.NewListener(logger, contractABIs...)
+
+			// example addresses
+			testnetContracts := []common.Address{
+				common.HexToAddress("0xEDEDB8ed37A43Fd399108A44646B85b780D85DD4"), // AVS
```

Comment:
This should use constants from the contracts-abi pkg. If the addresses change, this would be auto updated.